middleware/index.ts

```diff
@@ -91,11 +91,19 @@ export default function applySfFxMiddleware(request: any, state: any, resultArgs
         delete data.sfContext;
     }
 
+    // logger is the last element in the array
+    const logger: Logger = resultArgs[resultArgs.length-1];
+    if (!logger) {
+      throw new Error('Logger not provided in resultArgs from node system function');
```

Comment:
When will this happen?  Or just being protective?

Reply:
it is protective...as right now the contract is logger is passed in as the last element of the resultArgs array. 
I was originally wanted to get the element that's type Logger, but can't really do instanceof Logger... come to think of it... I could add `!logger.info && !logger.warn` check to be sure that's a logger object